src/components/game/chat-thread.tsx

```diff
@@ -82,6 +85,99 @@ export default function ChatThread(props: {
     (state) => state.gameData.room?.gameData.chat || []
   );
 
+  const players = useAppSelector(
+    (state) => state.gameData.room?.gameData.players
+  );
+
+  player?.avatar;
+  enum PlayerColors {
+    Blue = 'info.main',
+    Green = 'success.main',
+    Orange = 'warning.main',
+    Lavender = 'secondary.main',
+    Grey = 'text.secondary',
+    Red = 'error.main',
+  }
+  // eslint-disable-next-line prefer-const
+  let playerColorMap: Map<string, string> = new Map([]);
+
+  // eslint-disable-next-line prefer-const
+  let usedColors: Map<string, boolean> = new Map([
+    [PlayerColors.Green, false],
+    [PlayerColors.Lavender, false],
+    [PlayerColors.Orange, false],
+  ]);
+  //setting only 3 colors as we have 4 players max. Blue is reserved for Self and Grey is for System.
+
+  const GetUnusedColor = (): string => {
+    console.log('Used Colors', usedColors);
+
+    let retColor = PlayerColors.Red.toString();
+    // eslint-disable-next-line prefer-const
+    for (let myKey of usedColors.keys()) {
+      console.log(`Checking if key color ${myKey} is unused`);
+      if (usedColors.get(myKey) == false) {
+        console.log(`Found key color ${myKey} is unused and set to used`);
+
+        usedColors.set(myKey, true);
+        retColor = myKey;
+        break;
+      }
+    }
+    return retColor;
+  };
+  const GetMyColor = (id: string, isPlayer: boolean): string => {
+    if (id != '') {
+      if (!(id in playerColorMap)) {
+        if (isPlayer) {
+          playerColorMap.set(id, PlayerColors.Blue);
+        } else {
+          const unusedColor = GetUnusedColor();
+          playerColorMap.set(id, unusedColor);
+        }
+      }
+      return playerColorMap.get(id) as string;
+    }
+
+    return PlayerColors.Red;
+  };
+
+  const BorderedAvatar = styled(Avatar)`
+    border: 3px solid lightseagreen;
+  `;
+  const stringAvatar = (name: string, id: string) => {
+    if (name == undefined || name == '')
```

Comment:
can be shortened to if(!name) because both undefined and empty string are 'falsy'